client/src/screens/groups/GroupDetails.js

```diff
@@ -3,252 +3,89 @@ import React, { Component } from 'react';
-
-  return (
-    <View>
-      <View style={styles.headerContainer}>
-        <View style={styles.sectionTextContainer}>
-          <Text style={styles.headerName} numberOfLines={1}>{section}</Text>
-        </View>
-      </View>
-    </View>
-  );
-};
-SectionHeader.propTypes = {
-  section: PropTypes.string,
-};
-
-const MembersDisplay = (props) => {
-  const { id, displayName } = props.member;
-  return (
-    <TouchableHighlight key={id}>
-      <View style={styles.groupContainer}>
-        <Icon name="user" size={24} color="blue" />
-        <View style={styles.groupTextContainer}>
-          <View style={styles.groupTitleContainer}>
-            <Text style={styles.groupName} numberOfLines={1}>{displayName}</Text>
-          </View>
-        </View>
-      </View>
-    </TouchableHighlight>
-  );
-};
-MembersDisplay.propTypes = {
-  member: PropTypes.shape({
-    id: PropTypes.number.isRequired,
-    displayName: PropTypes.string.isRequired,
-  }),
-};
-
-const EventsDisplay = (props) => {
-  const { id, name, details } = props.event;
-  return (
-    <TouchableHighlight key={id}>
-      <View style={styles.eventContainer}>
-        <Icon name="bullhorn" size={24} color="blue" />
-        <View style={styles.eventTextContainer}>
-          <View style={styles.eventTitleContainer}>
-            <Text style={styles.eventName} numberOfLines={1}>{name}</Text>
-          </View>
-          <Text style={styles.eventText} numberOfLines={1}>{details}</Text>
-        </View>
-      </View>
-    </TouchableHighlight>
-  );
-};
-EventsDisplay.propTypes = {
-  event: PropTypes.shape({
-    id: PropTypes.number.isRequired,
-    name: PropTypes.string.isRequired,
-  }),
-};
-
-const ScheduleDisplay = (props) => {
-  const { id, name, details, startTime, endTime } = props.schedule;
-  let timeText = '';
-  if (startTime === 0 && endTime === distantFuture) {
-    timeText = 'Perpetual Schedule';
-  } else {
-    const startText = moment.unix(startTime).format('DD/MM/YY, HH:mm:ss');
-    const endText = moment.unix(endTime).format('DD/MM/YY, HH:mm:ss');
-    timeText = `${startText} - ${endText}`;
-  }
-  return (
-    <TouchableHighlight key={id}>
-      <View style={styles.scheduleContainer}>
-        <Icon name="calendar" size={24} color="blue" />
-        <View style={styles.scheduleTextContainer}>
-          <View style={styles.groupTitleContainer}>
-            <Text style={styles.scheduleName} numberOfLines={1}>{name} - {timeText}</Text>
-          </View>
-          <Text style={styles.scheduleText} numberOfLines={1}>{details}</Text>
-        </View>
-      </View>
-    </TouchableHighlight>
-  );
-};
-ScheduleDisplay.propTypes = {
-  schedule: PropTypes.shape({
-    id: PropTypes.number.isRequired,
-    name: PropTypes.string.isRequired,
-    details: PropTypes.string,
-    startTime: PropTypes.number.isRequired,
-    endTime: PropTypes.number.isRequired,
-  }),
+const GroupDetailsList = ({ items, render, networkStatus, refetch }) => (
```

Comment:
some tips to improve this:
* change this to a function and make it return the React components
  ```
  const GroupDetailsList = ({ items, render, networkStatus, refetch }) => {
    return <Container>...</Container>
  }
  ```
* define `renderItem` as a named function so that we can pull the complicated logic out:
  ```
  const renderItem = ({item}) => {
    ...
  }
  ```
* change from nested ternary (like the linter suggests this is probably a bad idea™) to if/else-if:
  ```
  if (mode === 'user') {
    return <UserListItem ... />
  ...
  ```
* I believe you should avoid using the name of components to select them, unless you're actually going to use the component itself. If you've gotta do translation anyway it's less confusing to the reader. If I see 'UserListItem' as a string I'm going to assume you're dynamically looking up the component name, which you're not.
* I think you should avoid using the name `render` because this name is usually used for the function that renders a component. I suggest `mode`.
* Put this component in it's own file separate from the screen.

Reply:
what do you gain by making something into a function and returning the component, compared to keeping it how ive got it... { with a return vs (... (I dont know what this is called.)

Reply:
it will enable you to do the other suggestions. Both are functions - the latter lets you have one or more statements whereas the former only allows a single expression.